projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-sender.php

```diff
@@ -566,15 +592,146 @@ public function test_do_sync_errors_if_read_only() {
 	 * Validate that WP_Error is returned in do_full_sync if JETPACK_SYNC_READ_ONLY is defined and true.
 	 */
 	public function test_do_full_sync_errors_if_read_only() {
-		\Automattic\Jetpack\Constants::set_constant( 'JETPACK_SYNC_READ_ONLY', true );
+		Constants::set_constant( 'JETPACK_SYNC_READ_ONLY', true );
 
 		$this->factory->post->create();
 		$response = $this->sender->do_full_sync();
-		\Automattic\Jetpack\Constants::clear_single_constant( 'JETPACK_SYNC_READ_ONLY' );
+		Constants::clear_single_constant( 'JETPACK_SYNC_READ_ONLY' );
 
 		$this->assertTrue( is_wp_error( $response ) );
 	}
 
+	/**
+	 * Test do_sync will spawn a dedicated Sync request when the corresponding setting is enabled.
+	 */
+	public function test_do_sync_spawns_dedicated_sync_request() {
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 1 ) );
+		$this->factory->post->create();
+
+		add_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ), 10, 3 );
+		$this->sender->do_sync();
+		remove_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ) );
+
+		$this->assertTrue( $this->dedicated_sync_request_spawned );
+
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 0 ) );
+	}
+
+	/**
+	 * Test do_sync will NOT spawn a dedicated Sync request when the corresponding setting is enabled if the Sync queue is empty.
+	 */
+	public function test_do_sync_will_not_spawn_dedicated_sync_request_with_empty_queue() {
+		$this->sender->get_sync_queue()->reset();
+
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 1 ) );
+
+		add_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ), 10, 3 );
+		$result = $this->sender->do_sync();
+		remove_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ) );
+
+		$this->assertFalse( $this->dedicated_sync_request_spawned );
+		$this->assertTrue( is_wp_error( $result ) );
+		$this->assertEquals( 'empty_queue_sync', $result->get_error_code() );
+
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 0 ) );
+	}
+
+	/**
+	 * Test do_sync will NOT spawn a dedicated Sync request when the corresponding setting is enabled if the Sync queue is locked.
+	 */
+	public function test_do_sync_will_not_spawn_dedicated_sync_request_with_locked_queue() {
+		$this->sender->get_sync_queue()->lock( 0 );
+
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 1 ) );
+
+		add_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ), 10, 3 );
+		$result = $this->sender->do_sync();
+		remove_filter( 'pre_http_request', array( $this, 'pre_http_sync_request_spawned' ) );
+
+		$this->assertFalse( $this->dedicated_sync_request_spawned );
+		$this->assertTrue( is_wp_error( $result ) );
+		$this->assertEquals( 'locked_queue_sync', $result->get_error_code() );
+
+		Settings::update_settings( array( 'dedicated_sync_enabled' => 0 ) );
+		$this->sender->get_sync_queue()->reset();
```

Comment:
If this test (or any other test that does the same) will fail, these lines would not be executed. It's probably better to move the clean up into the `setUp`/`tearDown` hooks

Reply:
Fully agree, thanks Yarik! I'm still working on those tests :) 
I have addressed this in https://github.com/Automattic/jetpack/pull/22506/commits/f7ddbed9b1867c61f0f1c7737f78d2625b9a996a